torch/nn/functional.py

```diff
@@ -1929,6 +1943,15 @@ def embedding_bag(input, weight, offsets=None, max_norm=None, norm_type=2,
                          "then it must have the same shape as the input ({})"
                          .format(per_sample_weights.shape, input.shape))
 
+    if padding_idx is not None:
+        if padding_idx > 0:
+            assert padding_idx < weight.size(0), 'Padding_idx must be within num_embeddings'
+        elif padding_idx < 0:
+            assert padding_idx >= -weight.size(0), 'Padding_idx must be within num_embeddings'
+            padding_idx = weight.size(0) + padding_idx
+    else:
+        padding_idx = -1
```

Comment:
Is there a good reason why `embedding` does this checking in the python level? It seems like it should be simple enough to do in C++ and avoid duplicating the checks in the `torch::` api version.

Reply:
I agree that it would be best for the `nn.functional` implementation to defer this check to the C++ level, so I'll make that change.

However, I think the `EmbeddingBag` and `Embedding` modules should still have this check because they call `_fill_padding_idx_with_zero()` upon initialization at the python level. That method simply does this:
```
    def _fill_padding_idx_with_zero(self) -> None:
        if self.padding_idx is not None:
            with torch.no_grad():
                self.weight[self.padding_idx].fill_(0)
```
So it would produce an error by itself if `self.padding_idx` it out of bounds. But I don't think the error produced by the indexing operation would be very clear from the user's perspective.

Reply:
Yeah, the module checks are fine. I was meaning the functional interface.